src/AI/CapPhaseAnalyzer.java

```diff
@@ -0,0 +1,454 @@
+    }
+
+    return null;
+  }
+
+  public GameAction findChainAction(GameMap gameMap, Unit unit, ArrayList<CapStop> chain)
+  {
+    XYCoord goal = chain.get(0).coord;
+
+    boolean includeOccupiedSpaces = false; // We assume that the area won't be cluttered
+    ArrayList<XYCoord> destinations = Utils.findPossibleDestinations(unit, gameMap, includeOccupiedSpaces);
+
+    // If we can get to our destination, go for it
+    if( destinations.contains(goal) )
+      return new CaptureLifecycle.CaptureAction(gameMap, unit, Utils.findShortestPath(unit, goal, gameMap));
+
+    Utils.sortLocationsByTravelTimeToCoord(unit, destinations, gameMap, goal);
+
+    for( XYCoord moveCoord : destinations )
+    {
+      Unit resident = gameMap.getLocation(moveCoord).getResident();
+      boolean spaceFree = null == resident;
+      if( !spaceFree )
+        continue; // I don't know how we got here, but bail
+
+      // Figure out how to get here.
+      GamePath movePath = Utils.findShortestPath(unit, moveCoord, gameMap);
+
+      return new WaitLifecycle.WaitAction(unit, movePath);
+    }
+    return null;
+  }
+
+  public CapPhaseAnalyzer(GameMap map, Army viewer)
+  {
+    contestedProps = new ArrayList<>();
+
+    ArrayList<XYCoord> props = new ArrayList<>();
+    HashMap<XYCoord, Commander> factoryOwnership = new HashMap<>();
+    for( int i = 0; i < map.mapWidth; i++ )
+    {
+      for( int j = 0; j < map.mapHeight; j++ )
+      {
+        Environment env = map.getEnvironment(i, j);
+        if(env.terrainType == TerrainType.FACTORY)
+        {
+          factoryOwnership.put(new XYCoord(i, j), map.getLocation(i, j).getOwner());
+        }
+        else if( env.terrainType.isProfitable() )
+        {
+          props.add(new XYCoord(i, j));
+        }
+      }
+    }
+
+    ArrayList<XYCoord> rightfulProps = new ArrayList<>();
+    ArrayList<XYCoord> rightfulFactories = new ArrayList<>();
+    ArrayList<XYCoord> startingFactories = new ArrayList<>();
+    // Fully calculate factory ownership based on who can cap each first
+    // Assumption: No contested factories
+    for( XYCoord neutralFac : factoryOwnership.keySet() )
+    {
+      final Commander currentOwner = factoryOwnership.get(neutralFac);
+      if( currentOwner != null )
+      {
+        if( viewer == currentOwner.army )
+          startingFactories.add(neutralFac);
+        continue; // Not actually neutral
+      }
+
+      int newOwnerDistance = Integer.MAX_VALUE;
+      Commander newOwner = null;
+      for( XYCoord ownedFac : factoryOwnership.keySet() )
+      {
+        final Commander owner = factoryOwnership.get(ownedFac);
+        if( owner == null )
+          continue; // Not yet owned
+
+        final Unit inf = new Unit(owner, owner.getUnitModel(UnitModel.TROOP));
+        inf.x = ownedFac.xCoord;
+        inf.y = ownedFac.yCoord;
+
+        final GamePath infPath = findFeasiblePath(inf, neutralFac, map);
+        if( null == infPath || infPath.getPathLength() < 1 )
```

Comment:
Hm. Maybe we should fix `Utils.findShortestPath()` so we don't need a double failure check on the outside.
Obviously not a super priority for this PR.